R/otp-setup.R

```diff
@@ -0,0 +1,290 @@
+
+
+  set_up <- try(system(text, intern = TRUE))
+
+  # Check for errors
+  if(grepl("ERROR",set_up[2]) | grepl("Error",set_up[1])){
+    message("Failed to build graph with message:")
+    message(set_up[2])
+  }else{
+    message("Graph built")
+
+  }
+  return(set_up)
+
+
+}
+
+
+#' Set up an OTP instance.
+#'
+#' OTP is run in Java and requires Java commands to be typed into the command line.
+#' The function allows the parameters to be defined in R and automatically passed to Java.
+#' This function sets up a local instance of OTP, for remote versions see documentation.
+#'
+#' The function assumes you have run otp_build_graph()
+#'
+#' @param dir A character string path to the directory containing the OTP .jar file and the OTP graph
+#' @param memory A positive integer. Amount of memory to assign to the OTP in GB, default is 2
+#' @param router A character vector for the name of the routers, must match with contents of dir, default "current"
+#' Only a single router is currently supported
+#' @param port A positive integer. Optional, default is 8080.
+#' @param secure_port A positive integer. Optional, default is 8081.
+#' @param analyist Logical. Should the analyist features be loaded? Default FALSE
+#' @param wait Logical, Should R wait until OTP has loaded before running next line of code, default TRUE
+#' @return
+#' This function does not return a value to R.
+#' If wait is TRUE R will wait until OTP is running (maximum of 5 minutes)
+#' @examples
+#' otp_setup("C:/otp","C:/data")
+#' otp_setup("C:/otp","C:/data", memory = 5, analyst = TRUE)
+#' @export
+otp_setup <- function(otp = NULL,
+                      dir = NULL,
+                      memory = 2,
+                      router = "current",
+                      port = 8080,
+                      securePort = 8081,
+                      analyst = FALSE,
+                      wait = TRUE)
+{
+  # Run Checks
+  jar_file <- otp_checks(otp = otp, dir = dir, router = router, graph = T)
+  memory <- floor(memory) # Can have fractions of GB
+
+
+  # Set up OTP
+  if(Sys.info()[['sysname']] == "Linux") {
+    message("You're on linux, this function is not yet supported")
+    stop()
+
+  }else if(Sys.info()[['sysname']] == "Windows"){
+
+    text <- paste0('java -Xmx',memory,'G -jar "',
+                   otp,'/',jar_file,
+                   '" --router ',router,
+                   ' --graphs "',dir,'/graphs"',
+                   ' --server --port ',port,
+                   ' --securePort ',securePort)
+
+    if(analyst){
+      text <- paste0(text," --analyst")
+    }
+    set_up <- try(system(text, intern = FALSE, wait = FALSE))
+
+
+  }else if(Sys.info()[['sysname']] == "Darwin"){
+    message("You're on Mac, this function is not yet supported")
+    stop()
+
+  }else{
+    message("You're on and unknow OS, this function is not yet supported")
+    stop()
+
+  }
+
+
+  # Check for errors
+  if(grepl("ERROR",set_up[2])){
+    message("Failed to build graph with message:")
```

Comment:
I think this message should be \"Failed to start OTP\" or something along those lines.